src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/state/Task.kt

```diff
@@ -0,0 +1,184 @@
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.UserRole.PROBATION_COM
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.ADDRESS_AND_RISK_CHECKS_IN_PROGRESS
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.APPROVED
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.AWAITING_ADDRESS_AND_RISK_CHECKS
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.AWAITING_DECISION
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.AWAITING_PRE_DECISION_CHECKS
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.AWAITING_REFUSAL
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.ELIGIBLE_AND_SUITABLE
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.INELIGIBLE_OR_UNSUITABLE
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.NOT_STARTED
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.OPTED_OUT
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.AssessmentStatus.TIMED_OUT
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.TaskStatus.COMPLETE
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.TaskStatus.IN_PROGRESS
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.TaskStatus.LOCKED
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.entity.state.TaskStatus.READY_TO_START
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.model.EligibilityStatus
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.service.policy.model.residentialchecks.ResidentialChecksStatus
+
+enum class Task {
+
+  ASSESS_ELIGIBILITY {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.eligibilityChecksStatus) {
+      EligibilityStatus.NOT_STARTED -> READY_TO_START
+      EligibilityStatus.IN_PROGRESS -> IN_PROGRESS
+      else -> COMPLETE
+    }
+  },
+
+  ENTER_CURFEW_ADDRESS {
+    override fun visibleForRoles() = setOf(PRISON_CA)
+    override fun taskStatus(assessment: Assessment): TaskStatus? = when (assessment.status) {
+      NOT_STARTED, ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS, INELIGIBLE_OR_UNSUITABLE -> LOCKED
+      ELIGIBLE_AND_SUITABLE -> READY_TO_START
+      AWAITING_ADDRESS_AND_RISK_CHECKS, ADDRESS_AND_RISK_CHECKS_IN_PROGRESS -> COMPLETE
+      OPTED_OUT -> IN_PROGRESS
+      else -> null
+    }
```

Comment:
We'll likely gradually start to move these over as required to be more based on the state of an assessment rather than the AssessmentStatus.

This is to cope with things like if a case goes past this state, we still need to show progress rather than the case being locked